### ISSUE TITLE
Don't enhance?

### ISSUE THREAD

**Issue by adamsilver** [adamsilver/adamsilver.io-jekyll#185]
Don't enhance: you don't need to.

Talk about Jeremy's rules about Enhance!

Page refresh and u don't need it. 

Don't need enhance reference enhance article that your yet to right. E.g. if u use page refresh u get to use real better indicator without any work. Once u enhance u may need to enhance all the things. U need a indicator. U might need animation. 

Yes animation is another layered enhancement but it's still something that's now on the radar. And what have u got.  U certainly haven't saved a round trip. Uve actually had to download a fuck load more js and run the fucker. Not to mention developmemt and test time etc and so on.  Thats code u got maintain forever.  

The best code is the code u don't have to write. Ajax doesn't save u a round trip. It's often used a reason to make it app like.

The extreme of this is single page apps. But have u built a website without js before and with. well oiled backend?

Users don't notice the refresh. Have u observered users using a single page app (booking bug) on a slow connection.  It's awful.

And whilst u can use it for little requests on more long lived pages it encourages people to use it for whole pages.  For example a contact us page with a form on it.  Yes u can do it with Ajax. But ur going to submit values to server. Then ur going render a new "successfully submitted" page. Ur better off just leaving it as a page refresh.  

Remember Jeremie's 1 2 3. U don't need it here.  But u do need the mindset he describes. If u can't see thru the design requirements. And if u don't have the experience. Then it will always seem better to do the Enhancement (as long as u have the baseline). But that is just not the case. 

On a recent project I'm on I've learnt something I thought I would never learn. Not having client side validation is not really a problem. Well designed forms (link to one thing per page) with excellent content and easy to use labels means a round trip rarely returns an error.  Plus u gotta hit the server ultimately anyway. 

No need to even tackle universal js and share validation on client and server (it's doable but certainly harder on this Java project).

1. If u embrace page refresh then u don't need to develop and enhancement in js and ajax. Save time. Save cost. No need to have an api that handles error states. Or js to handle timeouts.

2.  If u embrace the page refresh then it encourages u to practice lean development practices by delivering the simplest version first. Solving the problem as low down the stack as poss. Derek festherstone.

3. Then it encourages u to use a Progressive enhancement layered approach.

4. The user downloads ur site quicker. Less js to download, parse and execute.

5. The user gets to use their browsers native and accurate loading indicator.

6. It encourages good design practices.  And solve problems at the root. If u use ajax then there's a tendendency to shove everything on the page. That's bad for cognitive load and the performance of the page load and render etc.

7. It embraces the continuum of the web. Granted ajax is a feature that u can and should use when the right situation arises. But avoiding the page refresh is materially dishonest. The web is made of links. Use them they work. They are the super power of the web.

8. The browser is best at parsing and rendering html. If u use ajax then u gotta do this in js. Slower. More prone to error. Etc.

Why progressive Enhancement. It allows u not to enhance.
